### ISSUE TITLE
Polyfill WebRTC?

### ISSUE THREAD

**Issue by klarkc** [pavlobu/deskreen#90]
I was trying to run the secondary screen on my Xbox One browser, (Microsoft Edge 18). ~un~Surprisingly it did not worked, so I tested in my computer with the same version, the console reports that WebRTC is not supported. So I found that WebRTC is only fully compatible with Edge 79+ (The Chromium based). But there is an implementation called Object RTC that matches [this Edge version](https://caniuse.com/objectrtc). I was wondering if it's not possible to somehow add a polyfill to allow the client runs on old browsers. I found [this lib](https://github.com/Temasys/AdapterJS) that adds a common layer that enables the protocol on a large range of browsers.

**Comment by pavlobu** [comment]
@klarkc Hi! Thank you for sharing it here. Definitely worth trying! I was thinking there is no hope for old browsers. This lib may change a lot. But definitely something I'm going to do long after I do things that needed to be done first in app..

Any code contributions from your side or from someone who happen to come across this post are highly appreciated and welcome! 

Let's continue this discussion here in case any news and ideas about this solution.

Cheers,
Paul

**Comment by pavlobu** [comment]
release 1.0.4 now supports polyfills for fixing blank browser page issue.
Issues with WebRTC may still occur on old browsers.

**Comment by klarkc** [comment]
More findings, the library being used as wrapper to webRTC does not support ORTC, and it is [out of plans](https://github.com/feross/simple-peer/issues/400#issuecomment-446373946).

**Comment by pavlobu** [comment]
Alright. it will takesome time to remove simple-peer as a dependency.
But seems we definitely need to move that way to support old browsers

**Comment by klarkc** [comment]
[AdapterJS](https://github.com/Temasys/AdapterJS) is opensource (Apache license), but it's extensions (that are only needed for screen shares) are commercial products.

MS Edge Object RTC does not support `RTCDataChannel`, this is the [reason why simple-peer does not support MS Edge](https://github.com/feross/simple-peer/issues/400#issuecomment-445956732)

**Comment by klarkc** [comment]
For RTCDataChannel there is [this polyfill lib](https://github.com/ShareIt-project/DataChannel-polyfill)

**Comment by pavlobu** [comment]
@klarkc 
Simple peer supports wrtc option in constructor. 
[here is what in docs](https://github.com/feross/simple-peer#api) :

```
wrtc - custom webrtc implementation, mainly useful in node to specify in the wrtc package. Contains an object with the properties:

RTCPeerConnection
RTCSessionDescription
RTCIceCandidate
```

we can get rid from sending video quality through RTCDataChannel as I did that to learn how it works. It can be sent throught regular websocket message as other messages are sent now.

**Comment by gougou851129** [comment]
Will chrome 61 and safari9.5 be supported on the schedule?

**Comment by ibobak** [comment]
My old ipad4 is also not supported - I tried Chrome, Safari and Firefox. Is this the same kind of problem described here?

**Comment by marc1uk** [comment]
yep, probably. webrtc support (h.264 only) was only added to Safari in ipads [in iOS 11](https://www.webrtc-developers.com/webrtc-on-chrome-firefox-edge-and-others-on-ios/), which was first available on the fifth generation ipads (released in 2017).
From the article:
> Before that ... the only possibility was to build a native iOS application that embeds the WebRTC stack.

I have heard that Ericsson's Bowser browser did in fact embed such a webrtc stack and was at one point available on the app store for ipad as early as gen 2 (iOS8), but was subsequently removed at Apple's request (wtf apple?) and I can't find an IPA to sideload to try it. :disappointed:

**Comment by RyanBaig** [comment]
any updates?

**Comment by pythonr0ck** [comment]
If anyone wants to try building an ipa for Ericsson's Bowser browser for iOS, here it is:
https://github.com/EricssonResearch/bowser
